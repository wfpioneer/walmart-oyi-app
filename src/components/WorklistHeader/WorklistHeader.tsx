import React from 'react';
import { Text, View } from 'react-native';
import styles from './WorklistHeader.style';
import { strings } from '../../locales';

interface worklistHeaderPropsI {
  title: string;
  numberOfItems: number;
}

const WorklistHeader = (props: worklistHeaderPropsI) => (
  <View style={styles.container}>
    <Text style={styles.headerTitle}>
      { props.title }
    </Text>
    <Text style={styles.numberOfItems}>
      { `${props.numberOfItems} ${props.numberOfItems === 1 ? strings('GENERICS.ITEM') : strings('GENERICS.ITEMS')}`}
    </Text>
  </View>
);

export default WorklistHeader;
