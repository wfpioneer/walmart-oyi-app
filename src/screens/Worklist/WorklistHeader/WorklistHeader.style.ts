import { StyleSheet } from 'react-native';
import COLOR from '../../../themes/Color';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLOR.WHITE,
    alignItems: 'flex-start',
    justifyContent: 'center',
    paddingVertical: 5,
    marginTop: 10,
    paddingHorizontal: 15
  },
  headerTitle: {
    color: COLOR.GREY_700,
    fontSize: 14
  },
  numberOfItems: {
    color: COLOR.GREY_700,
    fontSize: 12
  }
});
