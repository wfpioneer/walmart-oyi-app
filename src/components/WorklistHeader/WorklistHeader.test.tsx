import React from 'react';
import { render } from '@testing-library/react-native';
import WorklistHeader from './WorklistHeader';

describe('WorklistHeader Component', () => {
  test('Test worklistHeader component snaphsot', () => {
    const {
      toJSON
    } = render(<WorklistHeader title="A1-12" numberOfItems={15} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
