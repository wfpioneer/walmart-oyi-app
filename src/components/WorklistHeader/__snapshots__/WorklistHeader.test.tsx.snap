// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorklistHeader Component Test worklistHeader component snaphsot 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 10,
      "paddingHorizontal": 15,
      "paddingVertical": 5,
    }
  }
>
  <Text
    style={
      {
        "color": "#616161",
        "fontSize": 14,
      }
    }
  >
    A1-12
  </Text>
  <Text
    style={
      {
        "color": "#616161",
        "fontSize": 12,
      }
    }
  >
    15 [missing "en.GENERICS.ITEMS" translation]
  </Text>
</View>
`;
